### PR TITLE
fix(skilavottord): Improve error handling when creating access and companies

### PR DESCRIPTION
**Pull request by birkirkristmunds**
[TS-931](https://dit-iceland.atlassian.net/browse/TS-931)
Attach a link to issue if relevant

## What

Improve the error handling when registering access and recycling companies

## Why

Prevent showing the user Unhandled Runtime Error in the UI

## Screenshots / Gifs

image

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for access control and recycling partner creation, suppressing redundant error messages while informing users through toast notifications.
	- Added checks to prevent duplicate access control entries based on national ID.
	- Improved clarity of error messages for recycling partner creation by including specific company IDs in conflicts.

- **Bug Fixes**
	- Improved clarity of error messages related to missing partner IDs in access control creation.

### CODE REVIEW

File: apps/skilavottord/web/screens/AccessControl/AccessControl.tsx [hunk 1]
```diff
@@ -145,6 +148,9 @@ const AccessControl: FC<React.PropsWithChildren<unknown>> = () => {
   const [updateSkilavottordAccessControl] = useMutation(
     UpdateSkilavottordAccessControlMutation,
     {
+      onError(_) {
+        // Hide Runtime error message. The error message is already shown to the user in toast.
+      },
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply apps/skilavottord/web/screens/AccessControl/AccessControl.tsx:153]: Errors are handled and shown to the user via toast in the onError function in the errorLink.ts

File: apps/skilavottord/web/screens/AccessControl/AccessControl.tsx [hunk 2]
```diff
@@ -135,6 +135,9 @@ const AccessControl: FC<React.PropsWithChildren<unknown>> = () => {
   const [createSkilavottordAccessControl] = useMutation(
     CreateSkilavottordAccessControlMutation,
     {
+      onError(_) {
+        // Hide Runtime error message. The error message is already shown to the user in toast.
+      },
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply apps/skilavottord/web/screens/AccessControl/AccessControl.tsx:140]: Errors are handled and shown to the user via toast in the onError function in the errorLink.ts

File: apps/skilavottord/web/screens/RecyclingCompanies/RecyclingCompanyCreate/RecyclingCompanyCreate.tsx [hunk 3]
```diff
@@ -50,6 +50,9 @@ const RecyclingCompanyCreate: FC<React.PropsWithChildren<unknown>> = () => {
   const [createSkilavottordRecyclingPartner] = useMutation(
     CreateSkilavottordRecyclingPartnerMutation,
     {
+      onError: (_) => {
+        // Hide Runtime error message. The error message is already shown to the user in toast.
+      },
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply apps/skilavottord/web/screens/RecyclingCompanies/RecyclingCompanyCreate/RecyclingCompanyCreate.tsx:55]: Errors are handled and shown to the user via toast in the onError function in the errorLink.ts

File: apps/skilavottord/web/screens/AccessControl/AccessControl.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply apps/skilavottord/web/screens/AccessControl/AccessControl.tsx:153]: Errors are handled and shown to the user via toast in the onError function in the errorLink.ts

**Review by veronikasif — APPROVED** [review]
lgtm 💯

**Pull request merged**